### DISCUSSION TITLE
Extension Management - How to get OperationId

### DISCUSSION THREAD

**Discussion by vividreportsryan** [microsoft/ALAppExtensions#21402] [Q&A]
Hello,

Various functions in the [Extension Management](https://github.com/microsoft/ALAppExtensions/tree/main/Modules/System/Extension%20Management) module require an OperationId as input. Where/how would I get that value?

Specifically, after calling UploadExtension(), I would like to check the status of the upload/installation by calling GetDeployOperationInfo() and/or GetDeploymentDetailedStatusMessage() but both of these functions require an OperationId as input.

Thank you.

**Comment by pri-kise** [comment]
Hi @vividreportsryan you should check out the table `"Extension Deployment Status"` and the procedure
```al
    /// <summary>
    /// Retrieves a list of all the Deployment Status Entries
    /// </summary>
    /// <param name="TempExtensionDeploymentStatus">Gets the list of all the Deployment Status Entries in a temporary record.</param>
    procedure GetAllExtensionDeploymentStatusEntries(var TempExtensionDeploymentStatus: Record "Extension Deployment Status" temporary)
    begin
        ExtensionOperationImpl.GetAllExtensionDeploymentStatusEntries(TempExtensionDeploymentStatus);
    end;
```

**Comment by vividreportsryan** [comment]
Thank you for that reply @pri-kise!

I think I see that I could iterate through the records of "Extension Deployment Status" and feed the OperationId through to GetDeployOperationInfo/GetDeploymentDetailedStatusMessage.

The question I would like to be able to answer is: which OperationId matches the UploadExtension() that I just started? 
I think I can approximate the answer by using logic; filtering the "Extension Deployment Status" records by Started_On (to only look at recent entries) and then querying GetDeployOperationInfo() to get the extension Version so that I can match what I just uploaded. This should give me a reasonable assurance that I am looking at the correct OperationId. Is there any definitive way to get the OperationId that corresponds to what was started by the UploadExtension() call?
